### PR TITLE
Update Model diagram

### LINKED ISSUE #156: Update UML diagram for Model

### PR DESCRIPTION
**Pull request by delishad21**
Currently, the model diagram in our developer guide is not accurate to how our updated application functions. Lets update it so it accurately reflects the structure of our program. 

This PR also updates .gitignore and gradle.build to fix compatibility issues with vscode

Resolves #156

### CODE REVIEW

File: docs/diagrams/ModelClassDiagram.puml [hunk 1]
```diff
@@ -37,18 +40,26 @@ UserPrefs .up.|> ReadOnlyUserPrefs
...
+
 
 Person -[hidden]up--> I
 UniquePersonList -[hidden]right-> I
 
-Name -[hidden]right-> Phone
+Name -[hidden]down-> Phone
 Phone -[hidden]right-> Address
 Address -[hidden]right-> Email
+Email -[hidden]right-> Matric
```
Comment (Zer0Legion) [inline docs/diagrams/ModelClassDiagram.puml:59]: Good addition Johan! I find the updated diagram very refreshing and insightful.

File: docs/diagrams/ModelClassDiagram.puml [hunk 2]
```diff
@@ -37,18 +40,26 @@ UserPrefs .up.|> ReadOnlyUserPrefs
 
 AddressBook *--> "1" UniquePersonList
 UniquePersonList --> "~* all" Person
-Person *--> Name
+Person *-l-> Name
 Person *--> Phone
 Person *--> Email
 Person *--> Address
-Person *--> "*" Tag
+Person *--> "0..1" Matric
```
Comment (jayllo-c) [inline docs/diagrams/ModelClassDiagram.puml:47]: Great job adding the multiplicity!

**Review by jayllo-c — COMMENTED** [review]
Nice changes!

**Review by Zer0Legion — COMMENTED** [review]
Good update. LGTM

**Comment by danielleloh** [comment]
the changes all look good to me :)

**Review by Pughal77 — COMMENTED** [review]
Nice addition to the UML diagram very succinct

**Pull request merged**